pkg/network/protocols/http/ebpf_gotls.go

```diff
@@ -270,6 +271,11 @@ func (p *GoTLSProgram) Start() {
 			}
 		}
 	}()
+
+	_ = util.WithAllProcs(p.procRoot, func(pid int) error {
+		p.handleProcessStart(uint32(pid))
+		return nil
+	})
```

Comment:
does it happen synchronously?

Reply:
Yes. `handleProcessStart` may start a goroutine to trigger a binary analysis in case of an unknown binary, but all of the process registration logic/looping over existing processes from the data in `/proc` is done synchronously.

Reply:
If that's the case, then consider running the process scanner in a go routine so it wont block the gotls start